rllib/examples/rl_modules/classes/inverse_dynamics_model_rlm.py

```diff
@@ -0,0 +1,252 @@
+
+    .. testcode::
+
+            import numpy as np
+            import gymnasium as gym
+            import torch
+
+            from ray.rllib.core import Columns
+            from ray.rllib.core.rl_module.rl_module import RLModuleConfig
+            from ray.rllib.examples.rl_modules.classes.inverse_dynamics_model_rlm import (  # noqa
+                InverseDynamicsModel
+            )
+
+            B = 10  # batch size
+            O = 4  # obs (1D) dim
+            A = 2  # num actions
+            f = 25  # feature dim
+
+            # Construct the RLModule.
+            rl_module_config = RLModuleConfig(
+                observation_space=gym.spaces.Box(-1.0, 1.0, (O,), np.float32),
+                action_space=gym.spaces.Discrete(A),
+            )
+            icm_net = InverseDynamicsModel(rl_module_config)
+
+            # Create some dummy input.
+            obs = torch.from_numpy(
+                np.random.random_sample(size=(B, O)).astype(np.float32)
+            )
+            next_obs = torch.from_numpy(
+                np.random.random_sample(size=(B, O)).astype(np.float32)
+            )
+            actions = torch.from_numpy(
+                np.random.random_integers(0, A - 1, size=(B,))
+            )
+            input_dict = {
+                Columns.OBS: obs,
+                Columns.NEXT_OBS: next_obs,
+                Columns.ACTIONS: actions,
+            }
+
+            # Call `forward_train()` to get phi (feature vector from obs), next-phi
+            # (feature vector from next obs), and the intrinsic rewards (individual, per
+            # batch-item forward loss values).
+            print(icm_net.forward_train(input_dict))
+
+            # Print out the number of parameters.
+            num_all_params = sum(int(np.prod(p.size())) for p in icm_net.parameters())
+            print(f"num params = {num_all_params}")
+    """
+
+    @override(TorchRLModule)
+    def setup(self):
+        # Get the IDM achitecture settings from the our RLModuleConfig's (self.config)
+        # `model_config_dict` property:
+        cfg = self.config.model_config_dict
+
+        feature_dim = cfg.get("feature_dim", 288)
+
+        # Build the feature model (encoder of observations to feature space).
+        layers = []
+        dense_layers = cfg.get("feature_net_hiddens", (256, 256))
+        # `in_size` is the observation space (assume a simple Box(1D)).
+        in_size = self.config.observation_space.shape[0]
+        for out_size in dense_layers:
+            layers.append(nn.Linear(in_size, out_size))
+            if cfg.get("feature_net_activation"):
+                layers.append(
+                    get_activation_fn(cfg["feature_net_activation"], "torch")()
+                )
+            in_size = out_size
+        # Last feature layer of n nodes (feature dimension).
+        layers.append(nn.Linear(in_size, feature_dim))
+        self._feature_net = nn.Sequential(*layers)
```

Comment:
Dumb question: did you want to show with this example also hwo to build a Torch module from scratch? We could simply use our predefined MLPs.

Reply:
Good point, but yes, I wanted to show a more bare-bones approach w/o using too many RLlib-related utilities. Using only torch makes this RLModule much more readable for new users.